src/commands/createWebApp/readAzConfig.ts

```diff
@@ -0,0 +1,62 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { WorkspaceFolder } from 'vscode';
+import { IAppServiceWizardContext } from 'vscode-azureappservice';
+import { cpUtils } from '../../utils/cpUtils';
+import { getSingleRootWorkspace } from '../../utils/workspace';
+import { IDeployWizardContext } from '../deploy/IDeployWizardContext';
+
+/**
+ * Takes any number of Azure config properties to read and returns their configured values.
+ * Azure CLI configuration docs: https://aka.ms/AA64syh
+ */
+export async function readAzConfig(wizardContext: IAppServiceWizardContext & Partial<IDeployWizardContext>, ...propertiesToRead: AzConfigProperty[]): Promise<AzConfig> {
+    const config: AzConfig = {};
+    const workspaceFolder: WorkspaceFolder | undefined = getSingleRootWorkspace(wizardContext);
+    const workspacePath: string | undefined = workspaceFolder ? workspaceFolder.uri.fsPath : undefined;
+
+    try {
+        const azGlobalConfigListing = <IAzConfigItem[] | undefined>JSON.parse(await cpUtils.executeCommand(undefined, undefined, 'az configure --list-defaults --output json'));
+        const azLocalConfigListing = <IAzConfigItem[] | undefined>JSON.parse(await cpUtils.executeCommand(undefined, workspacePath, 'az configure --list-defaults --output json --scope local'));
+
+        readAzConfigListing(wizardContext, propertiesToRead, azGlobalConfigListing, config);
+
+        // Default to local config values if they exist
+        readAzConfigListing(wizardContext, propertiesToRead, azLocalConfigListing, config);
+    } catch (error) {
+        // Azure CLI not installed so no config to read
```

Comment:
This comment isn't entirely accurate - errors could happen for a lot of reasons (for example maybe they're on an old version of the cli that doesn't support `az configure`)

What about something like this:
```
// Suppress errors - this functionality should not block creating a web app
```